### PR TITLE
fix(container/virtualized-list) : Allow sorting with header column in display table mode

### PR DESCRIPTION
**Pull request by Sylvain-Talend**
**What is the problem this PR is trying to solve?**
Currently, it is not possible to sort via connected List when clicking column header

**What is the chosen solution to this problem?**
It is now possible to sort with column header

**Please check if the PR fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/containers/src/List/List.container.js [hunk 1]
```diff
@@ -102,6 +102,13 @@ class List extends React.Component {
 			list: {
 				items,
 				columns: get(this.props, 'list.columns', []),
+				sort: {
+					field: state && state.sortOn,
+					isDescending: state && !state.sortAsc,
```
Comment (jsomsanith) [inline packages/containers/src/List/List.container.js]: Test if `state` is truthy is useless, it is never falsy.

File: packages/containers/src/List/List.container.js [hunk 2]
```diff
@@ -102,6 +102,13 @@ class List extends React.Component {
...
+					onChange: (event, data) => {
```
Comment (jsomsanith) [inline packages/containers/src/List/List.container.js]: `onChange: this.onSelectSortBy` should be sufficient, no need to create a new function

File: packages/containers/src/List/List.container.js (hunk 1 above)
Comment (Sylvain-Talend) [reply packages/containers/src/List/List.container.js]: :+1:

File: packages/containers/src/List/List.container.js (hunk 2 above)
Comment (Sylvain-Talend) [reply packages/containers/src/List/List.container.js]: :+1:

**Pull request merged**